doc/user-manual/language/cubical.lagda.rst

```diff
@@ -1,106 +1,930 @@
+   transp A i1 a = a
+
+This requires ``A`` to be constant for it to be well-typed.
+
+We can use ``transp`` to define regular transport:
+
+::
+
+  transport : ∀ {ℓ} {A B : Set ℓ} → A ≡ B → A → B
+  transport p a = transp (λ i → p i) i0 a
+
+By combining the transport and min operations we can define the
+induction principle for paths:
+
+::
+
+  J : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+        (d : P x refl) {y : A} (p : x ≡ y)
+      → P y p
+  J P d p = transport (λ i → P (p i) (λ j → p (i ∧ j))) d
+
+One subtle difference between paths and the propositional equality
+type of Agda is that the computation rule for ``J`` does not hold
+definitionally. If ``J`` is defined using pattern-matching as in the
+Agda standard library then this holds, however as the path types are
+not inductively defined this does not hold for the above definition of
+``J``. In particular, transport in a constant family is only the
+identity function up to a path which implies that the computation rule
+for ``J`` only holds up to a path:
+
+::
 
--------------------------
-The interval type (``I``)
--------------------------
+  transportRefl : ∀ {ℓ} {A : Set ℓ} (x : A) → transport refl x ≡ x
+  transportRefl {A = A} x i = transp (λ _ → A) i x
 
-The variable ``i`` in the definition of ``refl`` has type ``I``, which
-topologically corresponds to the `real unit interval <https://en.wikipedia.org/wiki/Unit_interval>`_.
-In a closed context, there are only two values of type ``I``: the two
-ends of the interval, ``i0`` and ``i1``::
+  JRefl : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+           (d : P x refl) → J P d refl ≡ d
+  JRefl P d = transportRefl d
 
-  a b : I
-  a = i0
-  b = i1
+Internally in Agda the ``transp`` operation computes by cases on the
+type, so for example for Σ-types it is computed elementwise. For path
+types it is however not yet possible to provide the computation rule
+as we need some way to remember the endpoints of the path after
+transporting it. Furthermore, this must work for arbitrary higher
+dimensional cubes (as we can iterate the path types). For this we
+introduce the "homogeneous composition operations" (``hcomp``) that
+generalize binary composition of paths to n-ary composition of higher
+dimensional cubes.
 
-Elements of the interval form a `De Morgan algebra <https://en.wikipedia.org/wiki/De_Morgan_algebra>`_,
-with minimum (``∧``), maximum (``∨``) and negation (``~``).
+
+Partial elements
+----------------
+
+In order to describe the homogeneous composition operations we need to
+be able to write partially specified n-dimensional cubes (i.e. cubes
+where some faces are missing). Given an element of the interval ``r :
+I`` there is a predicate ``IsOne`` which represents the constraint ``r
+= i1``. This comes with a proof that ``i1`` is in fact equal to ``i1``
+called ``1=1 : IsOne i1``. We use Greek letters like ``φ`` or ``ψ``
+when such an ``r`` should be thought of as being in the image of
```

Comment:
```suggestion
when such an ``r`` should be thought of as being in the domain of
```
I think that’s what you meant?

Reply:
What I want to say is that when we write φ the argument should be thought as of being in the face lattice (which is not free as `(i=0) /\ (i=1) = \bot`). The IsOne map goes from the interval to the face lattice.

Reply:
Ok, but from the point of view of Agda, `IsOne` goes from `I` to `Setω`, so it’s strange to say that `r` is in the image of `IsOne`, as `r` is of type `I`.

Reply:
Ah I see what you're saying. I'm always thinking in terms of the semantics. With this in mind I think "domain" is less confusing